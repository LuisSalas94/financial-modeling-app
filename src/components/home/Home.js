import React, { useState, useEffect } from 'react';
import { FcStatistics } from 'react-icons/fc';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCompaniesFromURL } from '../../redux/home/home';
import Company from '../company/Company';

const Home = () => {
  const companiesState = useSelector((state) => state.homeReducer);
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');

  useEffect(() => {
    dispatch(fetchCompaniesFromURL());
  }, []);

  return (
    <section className="company-container">
      <div className="item">
        <div className="header">
          <div className="title">
            <FcStatistics className="icon" />
          </div>
          <h1>Financial Data For Every Need</h1>
        </div>
        <input
          type="text"
          onChange={(e) => setSearch(e.target.value)}
          placeholder="Search your favorite company..."
          className="input"
        />
        <div className="section">
          {companiesState
            .filter((company) => company.name.toLowerCase().includes(search.toLowerCase()))
            .map((company) => (
              <Company
                key={company.symbol}
                symbol={company.symbol}
                name={company.name}
                change={company.change}
                price={company.price}
                changesPercentage={company.changesPercentage}
              />
            ))}
        </div>
      </div>
    </section>
  );
};

export default Home;
