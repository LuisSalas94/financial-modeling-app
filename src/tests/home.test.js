import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/home/Home';
import store from '../redux/configureStore';

const MockedHome = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  </Provider>
);

describe('Home Component', () => {
  it('renders succesfully', () => {
    render(<MockedHome />);
    const title = screen.getByText('Financial Data For Every Need');
    expect(title).toMatchSnapshot();
  });
});
