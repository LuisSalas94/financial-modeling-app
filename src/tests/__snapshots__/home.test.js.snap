// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders succesfully 1`] = `
<h1>
  Financial Data For Every Need
</h1>
`;
